### PR TITLE
Professorships | @wmui51

### PR DESCRIPTION
**Pull request by wmui51**
# READY FOR REVIEW

## Summary

  * New Endowed Positions content. React code parses JSON dataset comprised of ~1200 records. Results are displayed across 7 schools and 16 centers/institutes/programs pages. URL of these "filtered" results are dynamically generated from the JSON file. Search filters for results. Users can then click to see an individual record.

## Criticality - 8

  * This content needs to be available in production to allow for the decommissioning of the D7 professorships.stanford.edu website. Redirects from legacy will need to be created.

# Review Tasks
## Setup tasks and/or behavior to test

  1. Check out this branch

  2. Navigate to the pages of results by clicking on any of the schools/center/institutes. Note that the top-level landing page is a "standard" interior storyblok-managed page providing pointers to the filtered results. We will need to updated the links to point to dev/prod when appropriate. 

  3. Verify the ability to move between filtered sets via the drop down.

  4. Verify the ability to conduct a search and view results.

## Front End Validation

  * [x]  Is the markup using the appropriate semantic tags and passes HTML validation?

  * [x]  Cross-browser testing has been performed?

  * [ ]  Automated accessibility scans performed?

  * [ ]  Manual accessibility tests performed?

  * [ ]  Design is approved by @ user?

### Code

  * [ ]  Are the naming conventions following our standards?

  * [x]  Does the code have sufficient inline comments?

  * [ ]  Is there anything in this code that would be hidden or hard to discover through the UI?

  * [x]  Are there any [code smells](https://blog.codinghorror.com/code-smells/)?

  * [x]  Are tests provided? eg (unit, behat, or codeception)

### Code security

  * [x]  Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?

  * [x]  Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General

  * [x]  Is there anything included in this PR that is not related to the problem it is trying to solve?

  * [x]  Is the approach to the problem appropriate?

# Affected Projects or Products

  * This content will allow for the decommissioning of the D7 Professorships site.

# Associated Issues and/or People

  * JIRA ticket(s)

  * Other PRs

  * Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)

  * Anyone who should be notified? (`@mention` them here)

  @sherakama @nplowman @buttonwillowsix 

# Resources

  * [AMP Tool](https://stanford.levelaccess.net/index.php)

  * [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)

  * [HTML Validator](https://validator.w3.org/)

  * [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

**Comment by sherakama** [comment]
@wmui51 & @fglick 

I know this is a POC so I don't want to get ahead of ourselves but would you like an early review?

**Comment by wmui51** [comment]
@sherakama no need on the review for now, as there is a lot of redundant code to clean up first, so don't want to waste your time going through that now. But I did have some questions as we move forward on this.

- Can I mix some `Storyblok` components with my custom SPA stuff?
- For the routing, I currently am using Gatsby's file system route api, which is why I have 20+ new files in there for pages, do you have a routing suggestion as each page is pretty much identical and will change based on some params?
- I currently have a json fixture which is ~300K which is all the data for the various pages, so once loaded it actually goes pretty quick in terms of filtering. Is this OK or do you think we should move to something like `gatsby-json-transformer`?
- On filtering, I currently installed `fuse.js` to traverse the data for fuzzy search, is there a preferred plugin for `gatsby`, I don't want to install extra packages if we don't need to.

Thanks for checking in on this.

**Comment by sherakama** [comment]
@wmui51

Sounds good. Part of code review is a discussion on approach and practice. I'll see about lining up some help to have a look at what is there so far and to answer your questions.

**Comment by wmui51** [comment]
@sherakama got it, thanks for that, maybe we can arrange that for next week for you to take a look? i want to clean up a lot of the copy pasta stuff to make it easier to see what's going on.

**Comment by sherakama** [comment]
@wmui51 

It won't likely be me but I will grab someone to have a look and start the thread with you here. Don't worry about the duplication or early state code as we are very used to sharing and discussing work at the beginning stages.

**Comment by nplowman** [comment]
1. The hard-coded JSON content is ok for prototyping. But this should eventually be moved over to the CMS (Storyblok). The idea here would be to create a new component to represents the sortable tables. That way content editors can create new pages with tables form the CMS without requiring a developer to custom-code a new page each time. And content changes can be made without committing new code. Shea can help get you access to the Storyblok space for this site if you need it. 

This will also help you eliminate most or all of the code duplication across the 20+ pages currently in this pull request. 

Walking through how the codebase integrates with Storyblok might require a more detailed walkthrough, so may be more efficient to have a 30 minute call on that. 

2. For styling - we are using Tailwind, which uses Utility classes. I'd take a quick look at the documentation and other components in our code base to get a sense of how we use it. https://tailwindcss.com/docs/installation . Note that we are adding a prefix to the default Tailwind classes. So if you see a class like 'w-full' in the documentation, you'll need to use "su-w-full" in this codebase. The Tailwind config has also been customized for some properties like colors and font size. If you're using a code editor like VS Code you can install a Tailwind plugin that will auto-complete the class names for you, which is really helpful.

**Comment by sherakama** [comment]
Wilbert should already have access to the Storyblok space: https://app.storyblok.com/#!/me/spaces/122401/

One note, this is the one exception website we have that is NOT using tailwindcss. This project was one of the first and uses our SCSS which you can find here: https://github.com/SU-SWS/ood_giving_site/tree/dev/src/scss

**Comment by nplowman** [comment]
Looks like Storyblok has an add-on that provides a table field: https://www.storyblok.com/apps/table

**Comment by forrestglick** [comment]
@nplowman Thanks for reviewing the initial code from @wmui51. I agree it would be great to set up a call to discuss how best to integrate the code with storyblok. Would early next week work for you? If so, please suggest a few times. Thanks!

**Comment by nplowman** [comment]
@forrestglick Sure - here's a few times that would work for me:

Monday: 
- 10am PST
- 1pm PST
- 2pm PST

Tuesday
- 10am PST
- 3pm PST

**Comment by nplowman** [comment]
@wmui51 Sorry - I just noticed I missed your questions a few comments up when I was reviewing yesterday. We can talk through these more when we meet next week, but did want to provide some quick responses on some of those.

> Can I mix some Storyblok components with my custom SPA stuff?
Yes, and there's a couple approaches for this. A common way would be to create the page in Storyblok (instead of creating it in code in /src/pages), and then create a component in Storyblok to represent your SPA app that has fields for any configurations you want to pass. Then in Gatsby, we just have a mapping that will tell it to render your spa app in the page wherever you placed the Storyblok component. 

> For the routing, I currently am using Gatsby's file system route api, which is why I have 20+ new files in there for pages, do you have a routing suggestion as each page is pretty much identical and will change based on some params?

Yes - but will be good to first to discuss if it will make more sense to manage the pages from the Storyblok side instead. If we decide it makes more sense to keep it in code - you can use dynamic routing to set up a wildcard route. (e.g. /professorships/[slug]), so all of these pages would be managed from a single code file. You can then grab the argument from the URL and use it to change behaviors. 
https://www.gatsbyjs.com/docs/reference/functions/routing/#dynamic-routing 

> I currently have a json fixture which is ~300K which is all the data for the various pages, so once loaded it actually goes pretty quick in terms of filtering. Is this OK or do you think we should move to something like gatsby-json-transformer?

We'll probably want to see if this data could be moved in Storyblok. But if we decide to keep it in the codebase, I don't see a huge problem just loading it the way you are now. 

> On filtering, I currently installed fuse.js to traverse the data for fuzzy search, is there a preferred plugin for gatsby, I don't want to install extra packages if we don't need to.
We are using Algolia Search for the sitewide search page, which is going to be a lot more robust and capable than js libraries. You don't necessarily have to use that, but I can walk you through it when we meet if you'd be interested in using it. The main thing is we'd need to work out is how to index the data. We already have some webhook set up for indexing pages on Storyblok when they are created or updated, but would probably need to add something to the build process for indexing the SPA content. 

The sitewide search on giving.stanford.edu is pretty basic, but you can take a look at their demos to get a sense of what it can do: https://www.algolia.com/doc/guides/building-search-ui/resources/demos/react/

**Comment by forrestglick** [comment]
@nplowman thanks for suggesting a few times to meet. I spoke with @wmui51 and agreed Monday at 10am PST works best for us. Here's a zoom link for the discussion:
Join from PC, Mac, Linux, iOS or Android: https://stanford.zoom.us/j/4576049771?pwd=eE5iVVdkV0kxV1NFdlJ6c2ZOVGx4Zz09
    Password: 839273

**Comment by nplowman** [comment]
@forrestglick Thanks - I have added it to my calendar.

**Comment by nplowman** [comment]
@wmui51 Here's a working example of how to import the sitewide header and footer with content into a page:
```
import React from 'react';
import { graphql } from 'gatsby';
import CreateStories from '../utilities/createStories';

const ImportDemo = ({ data }) => {
  const header = {
    ...data.header,
    content: JSON.parse(data.header.content),
  };

  const footer = {
    ...data.footer,
    content: JSON.parse(data.footer.content),
  }

  return (
    <div>
      <CreateStories stories={[header]} />
      <div>Your unique content goes here.</div>
      <CreateStories stories={[footer]} />
    </div>
  );
};

export const query = graphql`
  query ImportDemo {
    header: storyblokEntry(
      full_slug: { eq: "global-components/headers/giving-site-default-header" }
    ) {
      id
      content
    }
    footer: storyblokEntry(
      full_slug: {
        eq: "global-components/global-footers/global-footer-cardinal-red"
      }
    ) {
      id
      content
    }
  }
`;

export default ImportDemo;
```

**Comment by nplowman** [comment]
@wmui51 Here's the pager component I mentioned from the Alumni repo: 
Pager Component: https://github.com/SU-SWS/saa_alumni/blob/dev/src/components/search/searchPager.js
Utility that handles the elipses placement: https://github.com/SU-SWS/saa_alumni/blob/dev/src/utilities/buildPager.js 

(I can send you the two files if you don't have access. 

You'll likely need to refactor a few things for your use case (styling for sure), but might save you some work.

**Comment by forrestglick** [comment]
@nplowman Wilbert has made solid progress on integrating this code with Gatsby/Storkblok environment. Thanks again for your previous guidance. I've created a Storyblok homepage for this content on Dev: https://dev.giving.stanford.edu/endowed-positions/ where we will manually manage the links to the sub-pages of content. In Storyblok, I can link to pages via internal or external urls via the radio button selector. In our case, the links are internal urls, but generated dynamically via Wilbert's code. The internal url "picker" doesn't see the sub-pages, and Storyblok doesn't allow for submission of external urls without a full https:// address. Any suggestions on how to proceed? Please let me know if you need more context. Thanks!

**Comment by nplowman** [comment]
> @nplowman Wilbert has made solid progress on integrating his code with Gatsby/Storyblok environment. Thanks again for your previous guidance. I've created a Storyblok homepage for this content on Dev: https://dev.giving.stanford.edu/endowed-positions/ where we will manually manage the links to the sub-pages of content. In Storyblok, I can link to pages via internal or external urls via the radio button selector. In our case, the links are internal urls, but generated dynamically via Wilbert's code. The internal url "picker" doesn't see the sub-pages, and Storyblok doesn't allow for submission of external urls without a full https:// address. Any suggestions on how to proceed? Please let me know if you need more context. Thanks!

Hmm. That's a good question. If the links were being managed through a structured component type, I'd say to just use a text field to just enter relative slugs. But I'm not really sure of a good workaround for the Link plugin in the WYSIWYG. 

@sherakama Are there any existing work-arounds we are using on other sites for this type of case?

If not... I have one idea to pitch.
I tried using the prefix "internal://" instead of "https://" for an external URL, and it passes validation. 
So for internal pages not managed through storybook, we could use URLs like "internal://my-page/test".
And in the Gatsby codebase, we could check for the "internal://" prefix before rendering it replace it with a "/".

**Comment by sherakama** [comment]
> @sherakama Are there any existing work-arounds we are using on other sites for this type of case?

No, we generally encourage that pages get created in Storyblok and components get created to place the specific functionality. You can think of the search page as an example. This way Storyblok is the canonical source of truth for all things. 

Your suggestion to add an 'internal://' protocol could work. I'd want to make sure that this pattern works everywhere so we would need a change to https://github.com/SU-SWS/ood_giving_site/blob/dev/src/components/partials/sbLink.js

**Comment by forrestglick** [comment]
@nplowman @sherakama Thank you both for your thoughts on how best to manage the internal links to /endowed-positions subpages. I like the idea of supporting an 'internal://' protocol, but also realize this might take a bit of time to develop and evaluate. In the immediate, I guess I could use absolute (external) urls to point to the subpages, changing to giving.stanford.edu (obviously), when we deploy to prod. We are nearing our end-of-month goal for release, so hoping to do a bit more development and clean up this week and submit for Dev deployment.

**Comment by forrestglick** [comment]
> READY FOR REVIEW
> 
> 
> # Summary
> 
>     * New Endowed Positions content. React code parses JSON dataset comprised of ~1200 records. Results are displayed across 7 schools and 16 centers/institutes/programs pages. URL of these "filtered" results are dynamically generated from the JSON file. Search filters for results. Users can then click to see an individual record.
> 
> 
> # Review By June 9, 2022
> 
> 
> # Criticality - 8
> 
>     * This content needs to be available in production to allow for the decommissioning of the D7 professorships.stanford.edu website. Redirects from legacy will need to be created.
> 
> 
> # Review Tasks
> ## Setup tasks and/or behavior to test
> 
>     1. Check out this branch
> 
>     2. Navigate to the pages of results by clicking on any of the schools/center/institutes. Note that the top-level landing page is a "standard" interior storyblok-managed page providing pointers to the filtered results. We will need to updated the links to point to dev/prod when appropriate. 
> 
>     3. Verify the ability to move between filtered sets via the drop down.
>
>     4. Verify the ability to conduct a search and view results.
> 
> 
> ## Front End Validation
> 
>     * [x]  Is the markup using the appropriate semantic tags and passes HTML validation?
> 
>     * [x]  Cross-browser testing has been performed?
> 
>     * [ ]  Automated accessibility scans performed?
> 
>     * [ ]  Manual accessibility tests performed?
> 
>     * [ ]  Design is approved by @ user?
> 
> 
> ## Backend / Functional Validation
> ### Code
> 
>     * [ ]  Are the naming conventions following our standards?
> 
>     * [ ]  Does the code have sufficient inline comments?
> 
>     * [ ]  Is there anything in this code that would be hidden or hard to discover through the UI?
> 
>     * [ ]  Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
> 
>     * [ ]  Are tests provided? eg (unit, behat, or codeception)
> 
> 
> ### Code security
> 
>     * [ ]  Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
> 
>     * [ ]  Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?
> 
> 
> ## General
> 
>     * [ ]  Is there anything included in this PR that is not related to the problem it is trying to solve?
> 
>     * [ ]  Is the approach to the problem appropriate?
> 
> 
> # Affected Projects or Products
> 
>     * This content will allow for the decommissioning of the D7 Professorships site.
> 
> 
> # Associated Issues and/or People
> 
>     * JIRA ticket(s)
> 
>     * Other PRs
> 
>     * Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
> 
>     * Anyone who should be notified? (`@mention` them here)
> 
>     @sherakama @nplowman @buttonwillowsix 
> 
> # Resources
> 
>     * [AMP Tool](https://stanford.levelaccess.net/index.php)
> 
>     * [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
> 
>     * [HTML Validator](https://validator.w3.org/)
> 
>     * [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

**Comment by sherakama** [comment]
@nplowman When you get a chance could you please provide another review for Wilbert. Thank you.

**Comment by sherakama** [comment]
@wmui51 Please ensure that all status checks are green or any issues are discussed before code review.

**Comment by wmui51** [comment]
@sherakama i was waiting to see in the code review about the 3 issues that codeclimate reported, but see that two are no longer there so will fix the third. I will also ask before submitting next time but will upload the image rather than ref it from the give site, thanks.

**Comment by wmui51** [comment]
@sherakama @nplowman everything is green now.

**Comment by nplowman** [comment]
Codewise, I think everything looks good. Nothing jumped out at me that I was concerned about.

I did see some odd behaviors when using the "Search by Keywords" functionality though.

- The format and styling of the results is completely different than what is shown when you use the "Filter by school" dropdown. (These style differences might be intentional - I just found it disorienting)
- The pager is missing for the "Search by Keyword" results. If you use a keyword like "faculty", the results list can be unreasonably long. 
- When I click on an item in the "Search by Keyword" results, I would expect it to take me to a separate details page. But instead, it seems to just filter the list down to that one item. (If there isn't a details page to link to, maybe it would make sense to just remove the click-ability altogether?)
- If you click an item multiple times, it keeps appending the "item" parameter to the URL over and over again. (e.g. https://deploy-preview-249--adapt-giving.netlify.app/endowed-positions/search?term=yossi&item=1&item=0&item=0&item=0&item=0)

Not quite sure if these things are bugs, or "work as designed" - just things that I found confusing as a user.

**Comment by wmui51** [comment]
@forrestglick I believe the first three we have discussed and decided that that functionality is ok for the intial launch as it matches the live site.

@nplowman good catch on the query params, i will fix the extra query params.

**Comment by wmui51** [comment]
@nplowman @forrestglick I have fixed the extra query params when on a filtered single result.

**Review by nplowman — APPROVED** [review]
I think I'm good with this, with the query param bug fix and the other items validated as working as expected (at least for initial launch).

**Comment by sherakama** [comment]
@yvonnetangsu,

Please have a review of this work for UI/UX alignment and please pass this along to SODA for review. 

Steps to test:

1. Navigate to https://deploy-preview-249--adapt-giving.netlify.app/endowed-positions
2. Click on the school of humanities and sciences url
3. Perform a search for 'Jon' 
4. Select 'School of Engineering' from the down down filter
5. Scroll down and select page 4 from the pager. 

Thank you!

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu,
> 
> Please have a review of this work for UI/UX alignment and please pass this along to SODA for review.
> 
> Steps to test:
> 
> 1. Navigate to https://deploy-preview-249--adapt-giving.netlify.app/endowed-positions
> 2. Click on the school of humanities and sciences url
> 3. Perform a search for 'Jon'
> 4. Select 'School of Engineering' from the down down filter
> 5. Scroll down and select page 4 from the pager.
> 
> Thank you!

Thanks, @sherakama! I'll get to this later today 😄

**Comment by yvonnetangsu** [comment]
@sherakama (cc @forrestglick @wmui51)

A few initial thoughts before sending this to our SODA friends 😄:

**For the landing page**
https://deploy-preview-249--adapt-giving.netlify.app/endowed-positions

1. If we could use unordered list `<ul>` instead of `<p>` to group the 2 sets of links (have each link as a `<li>` item), that'd be awesome. That way it's more semantic and accessible. Also, there will be no need to use a `<br>` after each item.
Screen Shot 2022-06-06 at 3 26 39 PM


**For the search results pages, e.g.**
https://deploy-preview-249--adapt-giving.netlify.app/endowed-positions/school-of-humanities-and-sciences

2. The down caret on the "Filter by school or area" dropdown looks a bit stretched horizontally on my browser (latest Chrome, MacOS)
Screen Shot 2022-06-06 at 3 05 32 PM

3. When I test using @sherakama 's steps to test, before hitting enter, the way the input fields are laid out gives me the impression that it's a drill down search, ie, if "School of Engineering" is selected, and if I enter "Jon" in "search by keyword", then I'm looking for Jon within the School of Engineering. It turns out that searching with a keyword would clear out the school filter. Maybe tweaking the UI, e.g., adding an "or" between the 2 fields would make it clearer that this isn't a drill down search?

Screen Shot 2022-06-06 at 3 10 07 PM

4. Seems like the search results for the keyword search could use a heading (similar to the results when using the filter by school or area dropdown)

5. It would be nice if the page content background color is su-bg-foggy-light (grey) so you don't get this awkward line at the bottom of the intro box here. The alternate row coloring on the tables might need a small styling tweak though (I think currently the white rows actually has transparent background instead of white)
Screen Shot 2022-06-06 at 3 05 16 PM

6. Are the individual results meant to be clickable when I use keyword search? Currently, if I search for "Jon", I was able to click on an entry and it brought me into a page like this
Screen Shot 2022-06-06 at 3 41 32 PM

7. Perhaps we could use the same pagination behavior as the site search? Currently it gets cut off on mobile when there are many pages (this is the school of Humanities and Sciences)

Screen Shot 2022-06-06 at 3 44 50 PM

**Review by yvonnetangsu — COMMENTED** [review]
Please see comment added and I've passed this along to the SODA team 😄

**Comment by forrestglick** [comment]
@yvonnetangsu Thanks for taking a look and providing the great feedback.
As you suggested, I've updated the landing page to use an unordered list with the "Schools" items: https://dev.giving.stanford.edu/endowed-positions/ I was hoping to avoid the bullets and wanted the font to be a bit larger, hence the prior approach, but I defer to your preferences. If this looks ok to you, I'll apply the changes to the second grouping of items (Centers, Institutes, and Programs). Let me know if you suggest any other changes on that landing page.
Wilbert will be able to address the other suggestions you've made for the sub-pages. Thanks again!

**Comment by yvonnetangsu** [comment]
@wmui51 @forrestglick cc @sherakama 
Clare from SODA was able to review this quickly, and here's a link to her Google doc with the results and her recommendations. If there's a time crunch, please prioritize fixing the "critical" and "major" issues.
https://docs.google.com/spreadsheets/d/1LcF0ai0cyMX6ElCq7nU3vpxhF-g6B9GmVB4zf8A_suI/edit?usp=sharing

I asked her to take a quick look at the landing page (https://deploy-preview-249--adapt-giving.netlify.app/endowed-positions), and she mentioned she would like to see the links grouped in `ul` as well.

**Comment by wmui51** [comment]
@yvonnetangsu I believe all SODA items and all of the items you mentioned should be addressed except for #3 and 6 from your list. For #3, after discussing with @forrestglick it was decided to leave that as is for now, and #6 is intended functionality (at least for launch).

Thanks for you the code review and let me know if I need to make any other changes.

File: src/components/endowed-positions/EndowedPositionsPagination.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            className={cx('search-pagination-item', {
+              ['search-pagination-item--current']: item + 1 === currentPage,
+              ['search-pagination-item--mobile-hidden']:
+                mobileDisplayArray.indexOf(item) === -1 && isMobile,
+            })}
+            key={item + 1}
+            {...ariaCurrent}
+          >
+            {item + 1 === currentPage ? (
+              <>{item + 1}</>
```
Comment (sherakama) [inline src/components/endowed-positions/EndowedPositionsPagination.js:71]:
```suggestion
              <span class="sr-only">page </span>{item + 1}</>
```

**Comment by sherakama** [comment]
Pager does not fit for tablet sized screens on the H&S group.

Screen Shot 2022-06-13 at 10 20 01 AM

**Comment by sherakama** [comment]
Clicking on the "Go" button does not set the focus on the new content. When tabbing the `skip to main content` link is not working because the anchor is missing. 

Screen Shot 2022-06-13 at 10 30 03 AM

**Comment by sherakama** [comment]
@forrestglick & @wmui51 

I see a few more bugs in this work and have left comments. I will ask that you fix the content focus issue as that was marked as a `major` item from SODA.

Thanks.

**Comment by wmui51** [comment]
@sherakama i was wondering for the content focus issue, I think I have it so on change the content should `scrollIntoView()` and getting `focus()`, but the content doesn't have a `tabindex` so I'm not sure the `focus()` actually works, so should I be putting an anchor link to land on the updated content area?

**Comment by yvonnetangsu** [comment]
@wmui51 Adding one more - the Go button looks a bit off on my iPhone Chrome browser. Just tested on iPhone Safari and it's the same way.
IMG_6575

**Comment by sherakama** [comment]
Anchor works fine for me. The focus and scroll do not seem to work for me, however, on the go button. 

https://share.vidyard.com/watch/rZsQXtj7jNaoFxUeCSytkr

**Comment by wmui51** [comment]
thanks @sherakama for the video. i think i found the issue. i realized that it was an issue that was discussed with @forrestglick for the landing pages of the schools (page 1), where if they come from the `/endowment-positions` landing page and land on the filtered schools page it would auto scroll them to the content and the user would never see the nav bar, unless they knew to scroll back up.

i have fixed that via additional query params, and will check it up tonight.

**Comment by wmui51** [comment]
@sherakama @yvonnetangsu i think i fixed the focus issue along with the other comments. thanks again for the review.

**Comment by yvonnetangsu** [comment]
> thanks @sherakama for the video. i think i found the issue. i realized that it was an issue that was discussed with @forrestglick for the landing pages of the schools (page 1), where if they come from the `/endowment-positions` landing page and land on the filtered schools page it would auto scroll them to the content and the user would never see the nav bar, unless they knew to scroll back up.
> 
> i have fixed that via additional query params, and will check it up tonight.

@wmui51 I wonder if this will be an issue too. I noticed that on the endowment positions search results pages, the `id="main-content"` is added where the search results begin. So if someone click a link and goes into the GSB results page for example and hit tab, the skip link bring them directly to the search results, which bypasses the H1 (which I think is ok here since it's the same as the landing page), but it would also bypass the whole search interface 🤔 

As a comparison, eg, on the Areas to Support page, the `id="main-content"` is added on the container that contains the H1 of the page
Screen Shot 2022-06-14 at 4 44 45 PM

I re-tested the focus again, and I'm still having issues. If I'm using only keyboard, after hitting "Go" button using the filter, or after a keyword search, hitting the tab key would bring up the skiplink (which means the focus is still at the top of the page even though the search results scroll into view). I suspect that we might have to add a tabindex="0" to the search results heading to make it focusable like you mentioned before. Let me ask Clare from SODA and see if she has any recommendations.

Thanks for fixing the button, pagination and other issues! 😄

**Comment by wmui51** [comment]
@yvonnetangsu I added a `tabIndex` which now gets focus, from what I found on the interwebs that seems to be an accepted way to do this. As for the use case you mentioned, I thought about that too and thought the best place to shift focus to would be where the results are since the dropdown never changes. I also thought about bringing that area into view but because of how tall the header is, it felt weird to load it in the middle as it seemed more like something loaded wrong.

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu I added a `tabIndex` which now gets focus, from what I found on the interwebs that seems to be an accepted way to do this. As for the use case you mentioned, I thought about that too and thought the best place to shift focus to would be where the results are since the dropdown never changes. I also thought about bringing that area into view but because of how tall the header is, it felt weird to load it in the middle as it seemed more like something loaded wrong.

Thank you @wmui51 ! I like this new change on the search results when using the dropdown - and Clare was able to review quickly and liked this approach also. Her only request is that, if you could please update the behavior for keyword search so after hitting enter, the focus ands on the results heading instead of the `ul`, then that'd be perfect. I think after this change we're GTG 👍🏼 Thank you for your patience!

**Comment by sherakama** [comment]
Thanks for the follow-up review @yvonnetangsu.

This is looking great. Nice work @wmui51. Just the one final request from Yvonne and then this is GTG. 

@forrestglick Are the SEO and Algolia indexing items next on your list?

**Comment by wmui51** [comment]
@yvonnetangsu @sherakama  thanks for looking again, i've just checked up the last request.

**Review by sherakama — APPROVED** [review]
GTG

**Comment by wmui51** [comment]
@sherakama before you merge, do you want me to rebase and squash the commits?

**Comment by sherakama** [comment]
@wmui51 Thanks for the offer but that won't be necessary. I can do that through the GitHub UI with `squash and merge` which I will be doing now.

**Pull request merged**